var/spack/repos/builtin/packages/sio/package.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Sio(CMakePackage):
+    """SIO is a persistency solution for reading and writing binary data in SIO
+    structures called record and block. SIO has originally been implemented as
+    persistency layer for LCIO.
+    """
+
+    url      = "https://github.com/iLCSoft/SIO/archive/v00-00-02.tar.gz"
+    homepage = "https://github.com/iLCSoft/SIO"
+    git      = "https://github.com/iLCSoft/SIO.git"
+
+    maintainers = ['vvolkl', 'tmadlener']
+
+    version('master', branch='master')
+    version('0.0.2', sha256='e4cd2aeaeaa23c1da2c20c5c08a9b72a31b16b7a8f5aa6d480dcd561ef667657')
+
+    def url_for_version(self, version):
+        """Translate version numbers to ilcsoft conventions.
+        in spack, the convention is: 0.1 (or 0.1.0) 0.1.1, 0.2, 0.2.1 ...
+        in ilcsoft, releases are dashed and padded with a leading zero
+        the patch version is omitted when 0
+        so for example v01-12-01, v01-12 ...
+
+        :param self: spack package class that has a url
+        :type self: class: `spack.PackageBase`
+        :param version: version
+        :type param: str
+        """
+        base_url = self.url.rsplit('/', 1)[0]
+        major = str(version[0]).zfill(2)
+        minor = str(version[1]).zfill(2)
+        # handle the different cases for the patch version:
+        # first case, no patch version is given in spack, i.e 0.1
+        if len(version) == 2:
+            url = base_url + "/v%s-%s.tar.gz" % (major, minor)
+            # a patch version is specified in spack, i.e. 0.1.x ...
+        elif len(version) == 3:
+            patch = str(version[2]).zfill(2)
+            # ... but it is zero, and not part of the ilc release url
+            if version[2] == 0:
+                url = base_url + "/v%s-%s.tar.gz" % (major, minor)
+            # ... if it is non-zero, it is part  of the release url
+            else:
+                url = base_url + "/v%s-%s-%s.tar.gz" % (major, minor, patch)
+        else:
+            print('Error - Wrong version format provided')
+            return
```

Comment:
I would raise an error here instead

Reply:
HI @adamjstewart, good suggestion. Is there any special type of error you would have in mind for this? I would have gone for an `ExtrapolationError` but I am not sure if there is another, better match.

Reply:
Actually, after looking into this again, I have realized that I can probably handle all cases that should come this way and there should not be any need for raising an error here.